File: Assets/ToLua/Lua/user/ui/black_market/model/black_market_rule_data.lua
local t = {}
t.data = {}
local item = {}
item.__index = item

local game_res_data = require('ui/game/model/game_res_data')

function item.New(table)
	local o = {}
	setmetatable(o,item)
	o.id = tonumber(table.id)
	o.rule_id = tonumber(table.rule_id)
	o.item = game_res_data.NewByString(table.item)
	o.scale = table.scale
  o.materials = {}
  local index = 1
  if table.cost1 ~= '-1' then
    o.materials[index] = game_res_data.NewByString(table.cost1)
    index = index + 1
  end
  if table.cost2 ~= '-1' then
    o.materials[index] = game_res_data.NewByString(table.cost2)
    index = index + 1
  end
  if table.cost3 ~= '-1' then
    o.materials[index] = game_res_data.NewByString(table.cost3)
    index = index + 1
  end
  if table.cost4 ~= '-1' then
    o.materials[index] = game_res_data.NewByString(table.cost4)
    index = index + 1
  end
  
	return o
end

function t.GetDataById(id)
	return t.data[id]
end

local function Start()
	local origin = dofile('market_rule')
	origin.ForEach(function(id,table)
		local newItem = item.New(table)
		t.data[id] = newItem
	end)
end

Start()
return t

File: Assets/Code/Logic/UI/BlackMarket/Model/BlackMarketRuleData.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;
using Common.Util;
using Logic.Game.Model;

namespace Logic.UI.BlackMarket.Model
{
	public class BlackMarketRuleData  
	{
		private static Dictionary<int, BlackMarketRuleData> _blackmarketRuleDataDictionary;
		
		public static Dictionary<int, BlackMarketRuleData> GetBlackMarketRuleDatas()
		{
			if (_blackmarketRuleDataDictionary == null)
			{
				_blackmarketRuleDataDictionary = CSVUtil.Parse<int, BlackMarketRuleData>("config/csv/market_rule", "id");
			}
			return _blackmarketRuleDataDictionary;
		}
		
		public static Dictionary<int, BlackMarketRuleData> BlackMarketDataRuleDictionary
		{
			get
			{
				if (_blackmarketRuleDataDictionary == null)
				{
					GetBlackMarketRuleDatas();
				}
				return _blackmarketRuleDataDictionary;
			}
		}
		public static BlackMarketRuleData GetBlackMarketRuleDataById(int id)
		{
			if(BlackMarketDataRuleDictionary.ContainsKey(id))
			{
				return BlackMarketDataRuleDictionary[id];
			}
			return null;
		}

		public GameResData itemData;
		public List<GameResData> materials = new List<GameResData>();

		[CSVElement("id")]
		public int id;

		[CSVElement("rule_id")]
		public int rule_id;

		[CSVElement("item")]
		public string itemString
		{
			set
			{
				itemData = new GameResData(value);

			}
		}

		[CSVElement("cost1")]
		public string cost1
		{
			set
			{
				if(!value.Equals("-1"))
					materials.Add(new GameResData(value));

			}
		}

		[CSVElement("cost2")]
		public string cost2
		{
			set
			{
				if(!value.Equals("-1"))
					materials.Add(new GameResData(value));
			}
		}


		[CSVElement("cost3")]
		public string cost3
		{
			set
			{
				if(!value.Equals("-1"))
					materials.Add(new GameResData(value));
			}
		}


		[CSVElement("cost4")]
		public string cost4
		{
			set
			{
				if(!value.Equals("-1"))
					materials.Add(new GameResData(value));
			}
		}


	}
}

